content/app/algo/three/traversing.js

```diff
@@ -0,0 +1,61 @@
+define("content/app/algo/three/traversing", function(require, exports, module) {
+
+	/**
+	 * Utility for traversing three structure.
+	 * It is type independent. Visitor should be
+	 * implemented for specific type.
+	 */
+	var Traversing = Class(Object, {
+
+		$statics : {
+
+			/**
+			 * Traverse element with provided visitor.
+			 * @param  {Object} 			element
+			 * @param  {AbstractVisitor} 	visitor
+			 */
+			traverse : function (element, visitor) {
+				if (element) {
+					visitor.visit(element);
+					var children = visitor.extractChildren(element);
```

Comment:
In real visitor elements accepts visitor. Here we are traversing DOM elements, and they are not designed for visiting. That's the reason why I created traversing class. Other options is to attach to all DOM elements accept method, and that is really big hack :) right?

As it is more appropriate that traversing gets root element trough constructor that visitor (if you want to traverse the same tree with 2 or more different visitors) it is not possible to make recursive method as private. So only option is to make static that will get both visitor and element (or to make private that will do the same thing).

I've removed traverseList, but it could be useful to split this functionality for situations when you have list of nodes that you want to traverse... right? Maybe recursion with 2 methods looks spaghetti to you, but its not ;) it's widely used (for example in simple parsers).